plugins/gaming.py

```diff
@@ -116,19 +116,63 @@ def dice(text, notice):
         return "{} ({})".format(total, ", ".join(rolls))
 
 
+choice_split_re = re.compile(r'\s*(?:, ?or|,|or)\s*', re.I)  # regex for splitting choices for choose command
+
+
 @hook.command
 def choose(text, event):
     """<choice1>, [choice2], [choice3], etc. - randomly picks one of the given choices
     :type text: str
     """
-    choices = re.findall(r'([^,]+)', text.strip())
+    choices = choice_split_re.split(text.strip())
+
     if len(choices) == 1:
-        choices = choices[0].split(' or ')
-        if len(choices) == 1:
+        event.notice_doc()
+        return
+
+    return random.choice([choice.strip() for choice in choices])
+
+
+# array of digits/characters indexed by int value
+val_char_map = [chr(x) for x in
+                [c for r in (range(ord('0'), ord('9') + 1), range(ord('a'), ord('z') + 1)) for c in r]]
+
+# map of digit/characters to int index ('9' = 9, 'A' = 10, etc)
+char_val_map = {x: idx for idx, x in enumerate(val_char_map)}
+
+
+@hook.command("chooser", "chooserange")
+def chooserange(text, event, notice):
+    """<a> to <b>|<a>-<b> - picks a random integer or character from the given range"""
+    choices = re.split(r'\s*(?:to|-)\s*', text)
+
+    if len(choices) == 2:
+        try:
+            if choices[0].isdigit() and choices[1].isdigit():
+                # integer range
+                return str(random.randint(int(choices[0]), int(choices[1])))
+
+            if len(choices[0]) == 1 and choices[1].isalpha() and len(choices[1]) == 1:
+                # 0-9 + A-Z
+                # get the integer index of the digits/characters
+                intval_arg0 = char_val_map.get(choices[0])
+                intval_arg1 = char_val_map.get(choices[1])
```

Comment:
This could be replaced with `int(choices[0], 36)` to parse the values as base36.